### PR TITLE
merge pages and app overlays

### PR DESCRIPTION
**Pull request by ForsakenHarmony**
### What

Merges the package into the internal overlay.

### Detailed Changes

* Move `@next/react-dev-overlay` into next package and rename the import paths.
* Fix the `getErrorSource` symbol issue, use `Symbol.for()` instead of `Symbol`
* Extra `getErrorSource` into single lib as it's being used in a lot of places, this will keep the same bundle size

Closes PACK-2261
Closes NEXT-1977

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: fefe05301704cff1eabf6f092ead2f69402daf40

`pnpm test test/integration/config-devtool-dev/test/index.test.js`
- devtool set in development mode in next config > should warn and revert when a devtool is set in development mode

<details>
<summary>Expand output</summary>

  ● devtool set in development mode in next config › should warn and revert when a devtool is set in development mode

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `devtool set in development mode in next config should warn and revert when a devtool is set in development mode 1`

    Snapshot: "pages/index.js (5:10) @ eval

      3 | export default function Index(props) {
      4 | useEffect(() => {
    > 5 |   throw new Error('this should render')
        |        ^
      6 | }, [])
      7 | return <div>Index Page</div>
      8 | }"
    Received: null

      40 |         // TODO: add win32 snapshot
      41 |       } else {
    > 42 |         expect(await getRedboxSource(browser)).toMatchInlineSnapshot(`
         |                                                ^
      43 | "pages/index.js (5:10) @ eval
      44 |
      45 |   3 | export default function Index(props) {

      at Object.toMatchInlineSnapshot (integration/config-devtool-dev/test/index.test.js:42:48)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| buildDuration | 11.7s | 15.1s | ⚠️ +3.5s |
| buildDurationCached | 6s | 7.6s | ⚠️ +1.6s |
| nodeModulesSize | 200 MB | 283 MB | ⚠️ +83.4 MB |
| nextStartRea..uration (ms) | 434ms | 462ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| 193.HASH.js gzip | 181 B | 182 B | N/A |
| 3f784ff6-HASH.js gzip | 53.4 kB | 53.4 kB | ✓ |
| 433-HASH.js gzip | 29 kB | 29 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 240 B | 242 B | N/A |
| main-HASH.js gzip | 31.8 kB | 31.8 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 98.6 kB | 98.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 183 B | 181 B | N/A |
| amp-HASH.js gzip | 504 B | 502 B | N/A |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 253 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.18 kB | 4.18 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | ✓ |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.4 kB | 3.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 485 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| index.html gzip | 528 B | 526 B | N/A |
| link.html gzip | 540 B | 539 B | N/A |
| withRouter.html gzip | 523 B | 522 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| edge-ssr.js gzip | 94 kB | 94 kB | N/A |
| page.js gzip | 148 kB | 149 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 625 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.5 kB | 37.5 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.07 kB | 2.07 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js hrmny/merge-overlays-pre-redesign | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 169 kB | 169 kB | ✓ |
| app-page-exp..prod.js gzip | 95.6 kB | 95.6 kB | ✓ |
| app-page-tur..prod.js gzip | 96.3 kB | 96.3 kB | ✓ |
| app-page-tur..prod.js gzip | 90.8 kB | 90.8 kB | ✓ |
| app-page.run...dev.js gzip | 142 kB | 142 kB | ✓ |
| app-page.run..prod.js gzip | 90.2 kB | 90.2 kB | ✓ |
| app-route-ex...dev.js gzip | 24.2 kB | 24.2 kB | ✓ |
| app-route-ex..prod.js gzip | 16.8 kB | 16.8 kB | ✓ |
| app-route-tu..prod.js gzip | 16.8 kB | 16.8 kB | ✓ |
| app-route-tu..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| app-route.ru...dev.js gzip | 23.6 kB | 23.6 kB | ✓ |
| app-route.ru..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.39 kB | 9.39 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.67 kB | 9.67 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.39 kB | 9.39 kB | ✓ |
| pages-turbo...prod.js gzip | 22 kB | 883 kB | ⚠️ +861 kB |
| pages.runtim...dev.js gzip | 22.6 kB | 883 kB | ⚠️ +861 kB |
| pages.runtim..prod.js gzip | 22 kB | 883 kB | ⚠️ +861 kB |
| server.runti..prod.js gzip | 49.7 kB | 49.7 kB | N/A |
| Overall change | 894 kB | 3.48 MB | ⚠️ +2.58 MB |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-turbo...time.prod.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>pages.runtime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>pages.runtime.prod.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: fefe05301704cff1eabf6f092ead2f69402daf40

**Review by huozhi — APPROVED** [review]

**Comment by ForsakenHarmony** [comment]
* `canary`
  * **#60899** <a href="https://app.graphite.dev/github/pr/vercel/next.js/60899?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ForsakenHarmony and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/next/src/client/components/react-dev-overlay/internal/components/hot-linked-text/index.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {
   MAGIC_IDENTIFIER_REGEX,
 } from '../../../../../../shared/lib/magic-identifier'
 
-const linkRegex = /https?:\/\/[^\s/$.?#].[^\s"]*/i
+const linkRegex = /https?:\/\/[^\s/$.?#].[^\s)'"]*/i
```
Comment (timneutkens) [inline packages/next/src/client/components/react-dev-overlay/internal/components/hot-linked-text/index.tsx:7]: This change looks strange, what's the difference?

Comment (huozhi) [reply packages/next/src/client/components/react-dev-overlay/internal/components/hot-linked-text/index.tsx:7]: @ForsakenHarmony is this intentional for sth? It’s from the 1st commit of moving folders

**Review by timneutkens — APPROVED** [review]

File: packages/next/src/client/components/react-dev-overlay/internal/components/hot-linked-text/index.tsx (hunk 1 above)
Comment (ForsakenHarmony) [reply packages/next/src/client/components/react-dev-overlay/internal/components/hot-linked-text/index.tsx:7]: This was a change done to the pages overlay, but not the app overlay before

Comment (ForsakenHarmony) [reply packages/next/src/client/components/react-dev-overlay/internal/components/hot-linked-text/index.tsx:7]: https://github.com/vercel/next.js/commit/05d887dc3bc31f18c3e630ea052251a63a4f202c

**Pull request merged**